qa/common/infer_util.py

```diff
@@ -137,21 +146,20 @@ def infer_exact(tester, pf, tensor_shape, batch_size,
                          for x in input1_array.reshape(input1_array.size)], dtype=object)
         input1_array = in1n.reshape(input1_array.shape)
 
-    # prepend size of string to input string data
-    if input_dtype == np.object:
-        input0_array_tmp = serialize_byte_tensor(input0_array)
-        input1_array_tmp = serialize_byte_tensor(input1_array)
-    else:
-        input0_array_tmp = input0_array
-        input1_array_tmp = input1_array
-
+    # prepend size of string to output string data
     if output0_dtype == np.object:
-        output0_array_tmp = serialize_byte_tensor(output0_array)
+        if batch_size == 1:
```

Comment:
Actually we don't need `batch_size` at all in `infer_exact` as there is not such notion in v2. Do we have to condition on `batch_size`?

Reply:
We do. Without batch_size we cannot know if the first dimension of the shape is the batch_size of if the model input does not have the batch dimension

Reply:
I am confused, I thought we already passing the full shape (i.e. include batch dimension if model support batching): https://github.com/NVIDIA/triton-inference-server/pull/1355#discussion_r414192810